### DISCUSSION TITLE
Using Prisma plug-in to automatically add all fields

### DISCUSSION THREAD

**Discussion by marceloverdijk** [hayes/pothos#362] [Ideas]
Would it be be possible to create an object type from a Prisma model with aal fields added automatically without specifying them all by hand?

Oc it would be nice to be able to override certain fields or exclude them (like a password field in an User object).

Ps I'm only interested in queries not mutations.

**Comment by hayes** [comment]
This is not something that currently exists, it may something I add through a new plugin in the future.   One of the core concepts behind pothos is to create a separating between your data and your public schema, so that your API is not tied directly the your database.  I may add something to support automatically creating object from your schema in the future, but my priority was to initially create something that was more flexible and decoupled.  I've got some work in progress an a plugin that will add more automatic type generation in #348 but it won't be ready for a while.

**Comment by marceloverdijk** [comment]
Thx for your feedback and I understand the rationale for the separation, which is general a good practice I fully agree.

Just to give you some background, my use case is that I have Sqlite database from which I generate a static website using Nextjs.
Coming from Gatsby which has an out-of-the-box GraphQL data layer, I'm trying to do the same with Nextjs, and hence my 1-to-1 mapping with the database.

**Reply by hayes** [reply]
Hey, just wanted to post an update here.  I am working on a prisma utils package for things that are not supported in the prisma plugin.  The idea is to make it easier to build your own tools to add these kinds of features that may not make sense as built in features.  Discussion is here: https://github.com/hayes/pothos/discussions/544  and writing a util that makes exposing all prisma fields easier might be in-scope for that package.
